### PR TITLE
[ztd-text] Add port

### LINKED ISSUE #23661: [New Port Request] ztd.text
Library name: ztd.text

Library description: A spicy text library for C++ that has the explicit goal of enabling the entire ecosystem to share in proper forward progress towards a bright Unicode future.

Source repository URL: https://github.com/soasis/text

Project homepage (if different from the source repository): https://ztdtext.readthedocs.io/en/latest/

Anything else that is useful to know when adding (such as optional features the library may have that should be included): Various build switches are documented at: https://ztdtext.readthedocs.io/en/latest/config.html

### PR DESCRIPTION
**Pull request by nickelpro**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes #23661

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?

UWP not supported due to: https://github.com/soasis/idk/issues/3. ~~CI baseline is updated~~ `"supports": "!uwp"` added to vcpkg.json

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
To the best of my ability, but this is my first port and I expect there will be mistakes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by nickelpro** [comment]
I'm unfamiliar with UWP and unsure what to do to make it happy, as the build failure is somewhat inscrutable. Open to suggestions, otherwise I'll add a UWP failure to baseline.txt

File: ports/ztd-text/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "ztd-text",
+  "version": "0.0.1",
+  "description": "A spicy text library.",
+  "homepage": "https://github.com/soasis/text",
+  "license": "Apache-2.0",
```
Comment (mathisloge) [inline ports/ztd-text/vcpkg.json:6]: `"supports": "!uwp",`
you can then remove the things from ci.baseline.txt

Comment (nickelpro) [reply ports/ztd-text/vcpkg.json:6]: Done

File: ports/ztd-text/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  REPO soasis/idk
+  REF 6173bb3d433a3dab65e8c356ef69441aa5a86275
+  SHA512 083711e9fc9b03aac34c5a7e182691a9bc9ee2d7c2d9ed2f55093423e2655f7a3829bd3e3db82bc42a78721795ca2d21a7b64c71f05e3c03ea3198f6228eb6f2
+  HEAD_REF main
+)
+
+vcpkg_from_github(
+  OUT_SOURCE_PATH SOURCE_PATH
+  REPO soasis/text
+  REF a562a6a2cb902e7ba0772419fedb32b3e7e8384c # latest
```
Comment (Cheney-W) [inline ports/ztd-text/portfile.cmake:20]:
```suggestion
  REF a562a6a2cb902e7ba0772419fedb32b3e7e8384c
```

File: ports/ztd-text/vcpkg.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "ztd-text",
+  "version": "0.0.1",
```
Comment (Cheney-W) [inline ports/ztd-text/vcpkg.json:3]:
```suggestion
  "version-date": "2022-03-20",
```
The value of the version-date should be the date you submitted this PR.

File: ports/ztd-text/portfile.cmake (hunk 2 above)
Comment (nickelpro) [reply ports/ztd-text/portfile.cmake]: Removed comment

**Review by Cheney-W — APPROVED** [review]

**Pull request merged**